packages/core/src/indexing/__tests__/query-filter-convert.test.ts

```diff
@@ -0,0 +1,116 @@
+import { jest } from '@jest/globals'
```

Comment:
Can you add tests with actual rows getting returned, checks for range support and other validation as described in https://github.com/ceramicnetwork/js-ceramic/pull/2839#discussion_r1233793048 please?